core/platform/datastore/cloud_datastore_services.py

```diff
@@ -89,8 +90,10 @@ def get_multi(keys):
     return ndb.get_multi(keys)
 
 
-def update_timestamps_multi(entities, update_last_updated_time=True):
-    # type: (Sequence[base_models.BaseModel], bool) -> None
+def update_timestamps_multi(
+        entities: Sequence['base_models.BaseModel'],
```

Comment:
Lets suppose :
```
class A:
  pass

class B(A):
  pass

def func(x: Sequence[A]): ...

func([A(), A()]) # valid
func([B(), B()]) # valid
```

Reply:
Thanks @hardikkat24 for the explanation! @vojtechjelinek is that a good reason to proceed with current approach?

Reply:
Discussed offline with @hardikkat24.

@vojtechjelinek Elsewhere we have used `TYPE_MODEL_SUBCLASS` in the return type only (which makes sense there to allow the function to return the class or any of its subclass). We have only one exception where `TYPE_MODEL_SUBCLASS` is used in argument type - https://github.com/oppia/oppia/blob/789b9ac1a038352a4b487abdeab6715081ec342f/core/platform/datastore/cloud_datastore_services.py#L109. And it actually makes sense also because we **do** want all the model subclasses to be accepted. Same is the case here too. So I am changing the annotation now.

Also `TYPE_MODEL_SUBCLASS` and 'base_models.BaseModel' are looking equivalent and mypy didn't throw errors,  because `List[TYPE_MODEL_SUBCLASS]` has been replaced with `Sequence['base_models.BaseModel']`. List in invariant (but the typevar is making it covariant) and Sequence is covariant in itself - making both the usages looks equivalent in our case. I am choosing the former one i.e. `List[TYPE_MODEL_SUBCLASS]` because it looks simpler to understand, makes its purpose obvious, and also because we don't use Sequence much in our codebase.